test/ng/directive/ngModelOptionsSpec.js

```diff
@@ -0,0 +1,768 @@
+        expect(formOptions.getOption('updateOn')).toEqual('blur');
+        expect(formOptions.getOption('updateOnDefault')).toEqual(false);
+        expect(inputOptions.getOption('updateOn')).toEqual('');
+        expect(inputOptions.getOption('updateOnDefault')).toEqual(true);
+
+        dealoc(container);
+      });
+
+      it('should inherit all unspecified options if the options object contains a `"*"` property with value "$inherit"', function() {
+        var container = $compile(
+                  '<div ng-model-options="{ allowInvalid: true, debounce: 100, updateOn: \'keyup\' }">' +
+                    '<form ng-model-options="{ updateOn: \'blur\', \'*\': \'$inherit\' }">' +
+                      '<input ng-model-options="{ updateOn: \'default\' }">' +
+                    '</form>' +
+                  '</div>')($rootScope);
+
+        var form = container.find('form');
+        var input = container.find('input');
+
+        var containerOptions = container.controller('ngModelOptions').$options;
+        var formOptions = form.controller('ngModelOptions').$options;
+        var inputOptions = input.controller('ngModelOptions').$options;
+
+        expect(containerOptions.getOption('allowInvalid')).toEqual(true);
+        expect(formOptions.getOption('allowInvalid')).toEqual(true);
+        expect(inputOptions.getOption('allowInvalid')).toEqual(false);
+
+        expect(containerOptions.getOption('debounce')).toEqual(100);
+        expect(formOptions.getOption('debounce')).toEqual(100);
+        expect(inputOptions.getOption('debounce')).toEqual(0);
+
+        expect(containerOptions.getOption('updateOn')).toEqual('keyup');
+        expect(containerOptions.getOption('updateOnDefault')).toEqual(false);
+        expect(formOptions.getOption('updateOn')).toEqual('blur');
+        expect(formOptions.getOption('updateOnDefault')).toEqual(false);
+        expect(inputOptions.getOption('updateOn')).toEqual('');
+        expect(inputOptions.getOption('updateOnDefault')).toEqual(true);
+
+        dealoc(container);
+      });
+
+      it('should make a copy of the options object', function() {
+        $rootScope.options = {updateOn: 'default'};
+        var inputElm = helper.compileInput(
+            '<input type="text" ng-model="name" name="alias" ' +
+              'ng-model-options="options"' +
+            '/>');
+        expect($rootScope.options).toEqual({updateOn: 'default'});
+        expect($rootScope.form.alias.$options).not.toBe($rootScope.options);
+      });
+
+      it('should be retrieved from an ancestor element containing an `ngModelOptions` directive', function() {
+        var doc = $compile(
+            '<form name="test" ' +
+                'ng-model-options="{ debounce: 10000, updateOn: \'blur\' }" >' +
+              '<input type="text" ng-model="name" name="alias" />' +
+            '</form>')($rootScope);
+        $rootScope.$digest();
+
+        var inputElm = doc.find('input').eq(0);
```

Comment:
I think that `find` does not return the object that we want (i.e. some collection thing) so we must convert it to get an instance.